src/impls/boxed.rs

```diff
@@ -1,46 +1,35 @@
-use no_std_io::io::Read;
+use no_std_io::io::{Read, Write};
 
 use alloc::boxed::Box;
 use alloc::vec::Vec;
 
-use bitvec::prelude::*;
-
 use crate::ctx::Limit;
-use crate::{DekuError, DekuReader, DekuWrite};
+use crate::reader::Reader;
+use crate::writer::Writer;
+use crate::{DekuError, DekuReader, DekuWriter};
 
 impl<'a, T, Ctx> DekuReader<'a, Ctx> for Box<T>
 where
     T: DekuReader<'a, Ctx>,
     Ctx: Copy,
 {
     fn from_reader_with_ctx<R: Read>(
-        reader: &mut crate::reader::Reader<R>,
+        reader: &mut Reader<R>,
         inner_ctx: Ctx,
     ) -> Result<Self, DekuError> {
         let val = <T>::from_reader_with_ctx(reader, inner_ctx)?;
         Ok(Box::new(val))
     }
 }
 
-impl<T, Ctx> DekuWrite<Ctx> for Box<T>
```

Comment:
We'd still want `DekuWriter` for `Box` to support:

```rust
#[derive(DekuWrite)]
struct MyStruct {
    field: Box<u8>,
}
```

Without it I'm getting:
```
the trait bound `Box<u8>: DekuWriter<_>` is not satisfied
```

Reply:
Good catch, should be good now